File: harness/validation/interactive-20260916/final-existing-js/website/src/site.js
function escapeHtml(value) {
  return String(value)
    .replaceAll("&", "&amp;")
    .replaceAll("<", "&lt;")
    .replaceAll(">", "&gt;")
    .replaceAll('"', "&quot;")
    .replaceAll("'", "&#39;");
}

export function render(items, category = "all") {
  const visibleItems = category === "all"
    ? items
    : items.filter(item => item.category === category);

  return "<ul>" + visibleItems
    .map(item => "<li>" + escapeHtml(item.title) + "</li>")
    .join("") + "</ul>";
}

File: harness/validation/interactive-20260916/final-existing-python/website/src/site.py
import html


def render(items, category="all"):
    """Render workshop titles, optionally filtering them by category."""
    visible_items = (
        items
        if category == "all"
        else (item for item in items if item.get("category") == category)
    )
    return "<ul>" + "".join(
        "<li>" + html.escape(str(item.get("title", ""))) + "</li>"
        for item in visible_items
    ) + "</ul>"
